tutorials/text/text_intro.py

```diff
@@ -93,3 +96,337 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Instead of making a list of all the tickalbels, we could have
+# used a `matplotlib.ticker.FormatStrFormatter` and passed it to the
+# ``ax.xaxis``
+
+fig, axs = plt.subplots(2, 1, figsize=(5, 3), tight_layout=True)
+axs[0].plot(x1, y1)
+axs[1].plot(x1, y1)
+ticks = np.arange(0., 8.1, 2.)
+# list comprehension to get all tick labels...
+formatter = matplotlib.ticker.StrMethodFormatter('{x:1.1f}')
+axs[1].xaxis.set_ticks(ticks)
+axs[1].xaxis.set_major_formatter(formatter)
+axs[1].set_xlim(axs[0].get_xlim())
+plt.show()
+
+#############################################################################
+# And of course we could have used a non-default locator to set the
+# tick locations.  Note we still pass in the tick values, but the
+# x-limit fix used above is *not* needed.
+
+fig, axs = plt.subplots(2, 1, figsize=(5, 3), tight_layout=True)
+axs[0].plot(x1, y1)
+axs[1].plot(x1, y1)
+formatter = matplotlib.ticker.FormatStrFormatter('%1.1f')
+locator = matplotlib.ticker.FixedLocator(ticks)
+axs[1].xaxis.set_major_locator(locator)
+axs[1].xaxis.set_major_formatter(formatter)
+plt.show()
+
+#############################################################################
+# The default formatter is the `matplotlib.ticker.MaxNLocator` called as
+# ``ticker.MaxNLocator(self, nbins='auto', steps=[1, 2, 2.5, 5, 10])``
+# The *steps* keyword is acceptable multiples that can be used for
+# tick values.  i.e. in this case, 2, 4, 6 would be acceptable ticks,
+# as would 20, 40, 60 or 0.2, 0.4, 0.6. However, 3, 6, 9 would not be
+# acceptable because 3 doesn't appear in the list of steps.
+#
+# ``nbins=auto`` uses an algorithm to determine how many ticks will
+# be acceptable based on how long the axis is.  The fontsize of the
+# ticklabel is taken into account, but the length of the tick string
+# is not (because its not yet known.)  In the bottom row, the
+# ticklabels are quite large, so we set ``nbins=4`` to make the
+# labels fit in the right-hand plot.
+
+fig, axs = plt.subplots(2, 2, figsize=(8, 5), tight_layout=True)
+axs = axs.flatten()
+for n, ax in enumerate(axs):
+    ax.plot(x1*10., y1)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.1f')
+locator = matplotlib.ticker.MaxNLocator(nbins='auto', steps=[1, 4, 10])
+axs[1].xaxis.set_major_locator(locator)
+axs[1].xaxis.set_major_formatter(formatter)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.5f')
+locator = matplotlib.ticker.AutoLocator()
+axs[2].xaxis.set_major_formatter(formatter)
+axs[2].xaxis.set_major_locator(locator)
+
+formatter = matplotlib.ticker.FormatStrFormatter('%1.5f')
+locator = matplotlib.ticker.MaxNLocator(nbins=4)
+axs[3].xaxis.set_major_formatter(formatter)
+axs[3].xaxis.set_major_locator(locator)
+
+plt.show()
+
+##############################################################################
+#  Finally, we can specify functions for the formatter using
+# `matplotlib.ticker.FuncFormatter`.
+
+
+def formatoddticks(x, pos):
```

Comment:
ternaries for the win: `return '%1.2f' % x if x % 2 else ''`

Reply:
Lets save that for the Advanced Python Tricks guide.  I want to be able to read this thing when its done ;-)